src/session/content/message_row/document.rs

```diff
@@ -0,0 +1,356 @@
+use glib::{clone, format_size};
+use gtk::glib::SignalHandlerId;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gio, glib, CompositeTemplate};
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use tdlib::enums::MessageContent;
+use tdlib::types::File;
+
+use crate::session::content::message_row::{
+    MessageBase, MessageBaseImpl, MessageIndicators, MessageLabel,
+};
+use crate::tdlib::{ChatType, Message, MessageSender};
+use crate::utils::parse_formatted_text;
+use crate::Session;
+
+use super::base::MessageBaseExt;
+
+mod imp {
+    use super::*;
+    use once_cell::sync::Lazy;
+    use std::cell::RefCell;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(resource = "/com/github/melix99/telegrand/ui/content-message-document.ui")]
+    pub(crate) struct MessageDocument {
+        pub(super) sender_color_class: RefCell<Option<String>>,
+        pub(super) bindings: RefCell<Vec<gtk::ExpressionWatch>>,
+        pub(super) status_handler_id: RefCell<Option<SignalHandlerId>>,
+        pub(super) click_handler_id: RefCell<Option<SignalHandlerId>>,
```

Comment:
```suggestion
        pub(super) click_handler_id: OnceCell<SignalHandlerId>,
```

Reply:
I need to use sync or unsync OnceCell?
![image](https://user-images.githubusercontent.com/63008755/189531201-d48d6759-69e1-4350-9477-76204ba99e20.png)

I think that GObjects is unsync but we have `once_cell::sync::Lazy` here`

Reply:
unsync is fine here

Reply:
Ok, I'm used unsync